packages/react-environment/src/api-providers/polkadot/polkadot-mixer-withdraw.ts

```diff
@@ -81,9 +80,19 @@ export class PolkadotMixerWithdraw extends MixerWithdraw<WebbPolkadot> {
     let done = false;
     let from = 0;
     let to = 511;
-    const leaves: Uint8Array[] = [];
-
+    const storage = this.inner.storage;
+    const mixerGroup = (await storage.get(String(treeId))) || {
+      0: { leaves: [] },
+    };
+    const mixerGroupForChain = mixerGroup[0];
+    const leaves: Uint8Array[] = mixerGroupForChain.leaves.map((leafStr) => {
+      return hexToU8a(leafStr);
+    });
+    // todo check this if the index starts from 1, or zero
+    from = Math.max(leaves.length - 1, 0);
+    to = from + 511;
     while (done === false) {
+      console.log(`Fetching leaves for treeId${treeId} from :${from} to ${to}`);
```

Comment:
I would use the proper logger that we have, I think there should be examples somewhere of its usage right @AhmedKorim ?